lib/connectors/mongodb/connector.rb

```diff
@@ -52,7 +52,7 @@ def initialize(configuration: {})
         @collection = configuration.dig(:collection, :value)
         @user = configuration.dig(:user, :value)
         @password = configuration.dig(:password, :value)
-        @direct_connection = configuration.dig(:direct_connection, :value)
+        @direct_connection = !!configuration.dig(:direct_connection, :value)
```

Comment:
`!!"foobar"` will also return `true`

Reply:
FYI here is how we do it in ent-search, which is pretty comprehensive: https://github.com/elastic/ent-search/blob/e4768dd6bbb17ddc7e75c272dd35285a79cf41df/config/initializers/string.rb#L80

Reply:
@mchernyavskaya fyi!

Reply:
I did it in a similar way here:
https://github.com/elastic/connectors-ruby/pull/337
But this way is a bit more permissive - it doesn't check for the correct false value. I think it's OK for us though. We might want to be a bit lax with these settings.

I also wasn't sure it was the best practice to monkey-patch string. But we can do it in connectors too if need be.